torch/csrc/jit/tensorexpr/cuda_half_support.h

```diff
@@ -38,6 +38,11 @@ class CudaHalfChecker : public IRMutator {
     return new Store(v->buf(), v->indices(), new_val, v->mask());
   }
 
+  const Expr* mutate(const HalfImm* v) override {
```

Comment:
So this will get us `float{v[i]} < float{0.}` right, but won't the dtype of the compare op still be `half`?  Does that cause problems?

Reply:
The default implementation of each operator in IRMutator will recreate it with the correct types if child nodes are modified.

I can modify the test to reach into the generated IR and check the dtype of the Max op if you like?